apps/frontend/src/lib/ws/index.ts

```diff
@@ -0,0 +1,101 @@
+import {BASE_API_WS} from "@/config";
+import {
+  WebSocketRequest,
+  WebSocketRequestTypes,
+  WebSocketResponseSchema,
+  WebSocketResponseTypes,
+  errorToString,
+  WebSocketResponse,
+  WebSocketErrorPayloadSchema,
+  WebSocketErrorPayload,
+  CreateMessageDataSchema,
+  CreateMessageData,
+  MessageData,
+  MapCreateMessageDataToMessageData
+} from "@types";
+import { getUserAndSession } from "@/lib/storage/localStorage/user";
+import { getUserSigPubKey } from "@/lib/user";
+import { storage } from "@/lib/storage";
+
+
+export const wsClient: WebSocket = new WebSocket(`${BASE_API_WS}`);
+
+wsClient.onopen = async () => {
+  console.log("Open ws server connection")
+
+  const user = await storage.getUser();
+  const session = await storage.getSession();
+
+  if (user) {
+    const sigPubKey: string = getUserSigPubKey(user);
+    if (sigPubKey && session) {
+      // WS connection is established earlier but need user public key for any requests
+      wsConnectUser(session.authTokenValue, sigPubKey);
+    }
+  } else {
+    console.warn("No user public signing key available, unable to establish websocket connection")
+  }
+};
+
+// @ts-expect-error : solve "TS2702: WebSocket only refers to a type, but is being used as a namespace here."
```

Comment:
Haven't figured out how to resolve this yet. I want to use native WebSockets and was having trouble importing the type.

Reply:
Ah "use client"